### PR TITLE
Add support for alpine 3.17, remove alpine 3.15

### PR DESCRIPTION
**Pull request by squizzi**
This PR adds support for alpine 3.17 and removes support for 3.15 to adhere to the policy of only supporting two major releases.

### CODE REVIEW

**Comment by squizzi** [comment]
Apparently alpine:3.17 is still in RC stages so I jumped the gun a bit on this, but it should be out officially by end of week.

**Comment by tianon** [comment]
Nice, thanks! :+1:

We should also ditch this entire `RUN` line (and associated comments) now that we don't need to deal with 3.15 anymore: :eyes:

https://github.com/docker-library/golang/blob/326acd5eed36954174ba8b3b6d0efda96087e18a/Dockerfile-linux.template#L13-L22

(I'm happy to take over and make this change if you would rather not do so!)

**Comment by squizzi** [comment]
> We should also ditch this entire RUN line (and associated comments) now that we don't need to deal with 3.15 anymore: eyes

How's that look?

File: Dockerfile-linux.template [hunk 1]
```diff
@@ -10,16 +10,6 @@ FROM alpine:{{ alpine_version }}
 
 RUN apk add --no-cache ca-certificates
```
Comment (tianon) [inline Dockerfile-linux.template:12]: Off-by-one; this empty line should go too :sweat_smile: 

```suggestion
```

(Again, happy to take care of this nit for you if you'd prefer!)

Comment (squizzi) [reply Dockerfile-linux.template]: :cry:

**Comment by tianon** [comment]
Nice thank you!  (and thanks for sticking with it!)

**Pull request merged**